src/models/generalized_mixture_reach_surface.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
```

Comment:
Thank you for writing this Ugur.  This is very complex code, and it's an important part of the project.